sunpy/net/helio/parser.py

```diff
@@ -217,18 +217,11 @@ def wsdl_retriever(service='HEC'):
     """
     def fail(): raise ValueError("No online HELIO servers can be found.")
     service_links = webservice_parser(service=service)
-    wsdl = None
-    wsdl_links = None
-    if service_links is None:
-        fail()
-    for link in service_links:
-        wsdl_links = taverna_parser(link)
-    if wsdl_links is None:
-        fail()
-    for end_point in wsdl_links:
-        if end_point is not None and link_test(end_point) is not None:
-            wsdl = end_point
-            break
-    if wsdl is None:
-        fail()
-    return wsdl
+    if service_links is not None:
+        for link in service_links:
+            wsdl_links = taverna_parser(link)
+            if wsdl_links is not None:
+                for end_point in wsdl_links:
+                    if end_point is not None and link_test(end_point) is not None:
+                        return end_point
+    fail()
```

Comment:
What caused this change?

Reply:
This is fix for this https://github.com/sunpy/sunpy/issues/2032 as I mentioned in the PR description.

Reply:
Will this change the results from #4357?

Reply:
Ah I missed that. I can't say I like how this block of code looks. But I don't have any better ideas.